src/timer/unix/SDL_systimer.c

```diff
@@ -192,8 +193,8 @@ void
 SDL_Delay(Uint32 ms)
 {
 #ifdef __EMSCRIPTEN__
-    if (emscripten_has_asyncify()) {
-        /* pseudo-synchronous pause */
+    if (emscripten_has_asyncify() && SDL_GetHintBoolean(SDL_HINT_EMSCRIPTEN_ASYNCIFY, SDL_TRUE)) {
```

Comment:
Since `emscripten_has_asyncify` is a JS call mabye put `SDL_GetHintBoolean` first to avoid doing it?

Reply:
I don't know. `emscripten_has_asyncify` is a static return, while SDL_GetHint/SDL_GetHintBoolean is a non-trivial function with a cost:
https://github.com/emscripten-ports/SDL2/blob/master/src/SDL_hints.c#L103